src/transformers/custom-elements-manifest/get-declarations.ts

```diff
@@ -0,0 +1,334 @@
+import * as tsModule from "typescript";
+import { isSimpleType, toSimpleType } from "ts-simple-type";
+import { ComponentDeclaration } from "../../analyze/types/component-declaration";
+import { getMixinHeritageClauses, getSuperclassHeritageClause } from "../../analyze/util/component-declaration-util";
+import { findParent, getNodeName } from "../../analyze/util/ast-util";
+import { getJsDoc } from "../../analyze/util/js-doc-util";
+import { getTypeHintFromType } from "../../util/get-type-hint-from-type";
+import { AnalyzerResult } from "../../analyze/types/analyzer-result";
+import * as schema from "./schema";
+import { TransformerContext } from "../transformer-context";
+import {
+	typeToSchemaType,
+	getSummaryFromJsDoc,
+	getParameterFromJsDoc,
+	getReturnFromJsDoc,
+	getReferenceFromHeritageClause,
+	getInheritedFromReference
+} from "./utils";
+
+/**
+ * Returns variables in an analyzer result
+ * @param result
+ * @param context
+ */
+function* getExportedDeclarationsFromResult(result: AnalyzerResult, context: TransformerContext): IterableIterator<schema.Declaration> {
+	// Get all export symbols in the source file
+	const symbol = context.checker.getSymbolAtLocation(result.sourceFile);
+	if (symbol == null) {
+		return;
+	}
+
+	const exports = context.checker.getExportsOfModule(symbol);
+
+	// Convert all export variables to VariableDocs
+	for (const exp of exports) {
+		const node = exp.valueDeclaration;
+
+		switch (exp.flags) {
+			case tsModule.SymbolFlags.BlockScopedVariable:
+			case tsModule.SymbolFlags.Variable: {
+				if (tsModule.isVariableDeclaration(node)) {
```

Comment:
what's the else case here?

Reply:
some of this is ported over from the implementation of the previous schema spec so i can't say for sure, but i think it'll purely be for the type guard. `node` needs narrowing as it'll be _any_ declaration. so a cast would work too but i suppose the if is safer